Source/WebCore/page/Chrome.cpp

```diff
@@ -196,7 +196,7 @@ Page* Chrome::createWindow(Frame& frame, const WindowFeatures& features, const N
         return nullptr;
 
     if (!features.noopener && !features.noreferrer) {
-        if (auto* oldSessionStorage = m_page.sessionStorage(false))
+        if (auto* oldSessionStorage = m_page.sessionStorage(m_page.mainFrame().document()->topOrigin(), false))
```

Comment:
It's probably worth null checking the document just to not introduce a new (and probably rare) crash.

Reply:
Added, and factored out `document`, as well.